vmm/src/device_manager.rs

```diff
@@ -449,6 +449,7 @@ impl DeviceRelocation for AddressManager {
         region_type: PciBarRegionType,
     ) -> std::result::Result<(), std::io::Error> {
         match region_type {
+            #[cfg(target_arch = "x86_64")]
```

Comment:
Rather than doing this (and making the field in the enum conditional) I think the code inside here should be architecture conditional and return an error on aarch64 to say that I/O regions are not supported.